server/app/mutations/grid_certificates/request_certificate.rb

```diff
@@ -17,56 +16,7 @@ class RequestCertificate < Mutations::Command
     end
 
     def validate
-      self.domains.each do |domain|
-        domain_authz = get_authz_for_domain(self.grid, domain)
-
-        if domain_authz
-          if domain_authz.authorization_type == 'dns-01'
-            # Check that the expected DNS record is already in place
-            unless validate_dns_record(domain, domain_authz.challenge_opts['record_content'])
-              add_error(:dns_record, :invalid, "Expected DNS record not present for domain #{domain}")
-            end
-          end
-        else
-          add_error(:authorization, :not_found, "Domain authorization not found for domain #{domain}")
-        end
-
-      end
-    end
-
-    def verify_domain(domain)
-      domain_authorization = get_authz_for_domain(self.grid, domain)
-      challenge = le_client.challenge_from_hash(domain_authorization.challenge)
-      if domain_authorization.state == :created
-        info "requesting verification for domain #{domain}"
-        success = challenge.request_verification
-        if success
-          domain_authorization.state = :requested
-          domain_authorization.save
-        else
-          add_error(:request_verification, :failed, "Requesting verification failed")
-        end
-      end
-
-      wait_until!("domain verification for #{domain} is valid", interval: 1, timeout: 30, threshold: 10) {
-        challenge.verify_status != 'pending'
-      }
-
-      case challenge.verify_status
-      when 'valid'
-        domain_authorization.state = :validated
-      when 'invalid'
-        domain_authorization.state = :invalid
```

Comment:
Can't use `:invalid` as a mongoid enum valid, it will result in an `invalid?` method on the model that overrides some mongoid API:

```
  1) Rpc::ServicePodSerializer#to_hash [:secrets] with a validated TLS-SNI-01 domain authorization does not include the challenge cert as a secret
     Failure/Error: GridDomainAuthorization.create!(grid: grid,
     ArgumentError:
       wrong number of arguments (given 1, expected 0)
     # (eval):1:in `invalid?'
     # ./vendor/bundle/ruby/2.3.0/gems/mongoid-5.2.1/lib/mongoid/persistable/creatable.rb:115:in `prepare_insert'
     # ./vendor/bundle/ruby/2.3.0/gems/mongoid-5.2.1/lib/mongoid/persistable/creatable.rb:23:in `insert'
     # ./vendor/bundle/ruby/2.3.0/gems/mongoid-5.2.1/lib/mongoid/persistable/creatable.rb:180:in `block in create!'
     # ./vendor/bundle/ruby/2.3.0/gems/mongoid-5.2.1/lib/mongoid/threaded/lifecycle.rb:161:in `_creating'
     # ./vendor/bundle/ruby/2.3.0/gems/mongoid-5.2.1/lib/mongoid/persistable/creatable.rb:175:in `create!'
     # ./spec/serializers/rpc/service_pod_serializer_spec.rb:254:in `block (5 levels) in <top (required)>'
```

Reply:
This is actually broken, but the `save` fails silently:  #3044